hack/ci-test.sh

```diff
@@ -29,6 +29,7 @@ GINKGO_EXTRA_ARGS=${GINKGO_EXTRA_ARGS:-""}
 # Ensure that some home var is set and that it's not the root.
 # This is required for the kubebuilder cache.
 export HOME=${HOME:=/tmp/kubebuilder-testing}
+export IS_TEST="true"
```

Comment:
I worry that this will be fragile, it means you can only actually run the tests via this script or by happening to know that we need to set this variable.

I'd prefer a solution in the units that doesn't rely on some external factor like this else it will make it hard to work locally